src/Symfony/Component/Config/Definition/EnumNode.php

```diff
@@ -78,6 +84,10 @@ protected function finalizeValue(mixed $value): mixed
     {
         $value = parent::finalizeValue($value);
 
+        if ($this->enumFqcn && !$value instanceof \BackedEnum) {
```

Comment:
You allow `$value` to be any `BackedEnum` while in fact it should be an instance of `$this->enumFqcn`. Also, if I haven't missed anything, you did not test the case that we receive an already hydrated enum here.